### PR TITLE
master deps 2024 09 06 13 40 04 783048

### PR DESCRIPTION
**Pull request by craigcomstock**

### CODE REVIEW

**Comment by craigcomstock** [comment]
@cf-bottom jenkins no test

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11187/)

 [NO TESTS]

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11187/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11187/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11191/)

 [NO TESTS]

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11191/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11191/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins no test

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11193/)

 [NO TESTS]

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11193/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11193/

**Comment by craigcomstock** [comment]
mingw retry [Build Status](https://ci.cfengine.com/job/pr-pipeline/11194/)

**Comment by craigcomstock** [comment]
retry mingw build with fixed patch: [Build Status](https://ci.cfengine.com/job/pr-pipeline/11196/)

**Comment by craigcomstock** [comment]
@cf-bottom jenkins, think I'm ready for the tests

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11198/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11198/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11198/

**Comment by craigcomstock** [comment]
rhel7hub seems like a flake failure, rebuild: [Build Status](https://ci.cfengine.com/job/pr-pipeline/11199/)

**Comment by craigcomstock** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11203/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11203/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11203/

**Comment by craigcomstock** [comment]
seems static-checks job is flaky? Some error with assembler saying no such pseudo-op .uleb...

[Build Status](https://ci.cfengine.com/job/pr-pipeline/11204/)

**Comment by craigcomstock** [comment]
jenkins checkboxes aren't clear, didn't run static-checks, retry again: [Build Status](https://ci.cfengine.com/job/pr-pipeline/11205/)

**Comment by craigcomstock** [comment]
@cf-bottom jenkins, full run again please, I think we should be "good"

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11206/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11206/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11206/

**Comment by craigcomstock** [comment]
see if the perl upgrade works. the build was working before this change so :crossed_fingers: [Build Status](https://ci.cfengine.com/job/pr-pipeline/11207/)

**Comment by craigcomstock** [comment]
check all platforms including exotics, I suspect exotics might have some problems?
[Build Status](https://ci.cfengine.com/job/pr-pipeline/11213/)

**Comment by craigcomstock** [comment]
Retry with openssl patch fix. Not sure why it wasn't failure elsewhere :thinking:
[Build Status](https://ci.cfengine.com/job/pr-pipeline/11217/)

**Comment by craigcomstock** [comment]
Better patch, previous one didn't compile :) [Build Status](https://ci.cfengine.com/job/pr-pipeline/11218/)

**Comment by craigcomstock** [comment]
@cf-bottom jenkins full run, thanks.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11220/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11220/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11220/

**Comment by craigcomstock** [comment]
oops, I need exotics, @cf-bottom jenkins with exotics please

**Comment by craigcomstock** [comment]
retrying just ubu22hub on arm without tests: [Build Status](https://ci.cfengine.com/job/pr-pipeline/11223/)

**Comment by craigcomstock** [comment]
@cf-bottom jenkins with exotics

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/11238/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/11238/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-11238/

**Comment by craigcomstock** [comment]
hpux failure is seen by me before and now a ticket: https://northerntech.atlassian.net/browse/ENT-12366

**Review by larsewi — APPROVED** [review]

File: deps-packaging/openssl/source [hunk 1]
```diff
@@ -1 +1 @@
-https://www.openssl.org/source/
+https://github.com/openssl/openssl/releases/download/openssl-3.3.2/
```
Comment (vpodzime) [inline deps-packaging/openssl/source:1]: Please explain this in the commit message.

File: build-scripts/install-dependencies [hunk 2]
```diff
@@ -40,13 +40,20 @@ check_and_install_perl()
...
         echo "$PERL version is $PERL_MINOR_VERSION," \
-             "too old, installing new one from source"
-
-        ### BUILD PERL
+             "too old"
+        PERL_OK="no"
+    fi
+    if ! perl -e 'use List::Util qw(pairs);'; then
+        echo "$PERL has List::Util that does not export pairs. Needs to be at least version 1.29 for OpenSSL version 3.3.2."
+        PERL_OK="no"
```
Comment (vpodzime) [inline build-scripts/install-dependencies:53]: This change deserves a better commit message, IMHO.

File: deps-packaging/openssl/0007-Avoid-using-Interlocked-64-Windows-functions.patch [hunk 3]
```diff
@@ -9,15 +9,15 @@ x86. There are already implementations of replacements for other
 functions, such as InterlockedOr64(). Apply the same approach to
 fix the errors.
 
-Inspired by openssl/openssl#24326.
+Inspired by https://github.com/openssl/openssl/pull/24326.
```
Comment (vpodzime) [inline deps-packaging/openssl/0007-Avoid-using-Interlocked-64-Windows-functions.patch:12]: Private commit referenced in the commit message for this change.

File: deps-packaging/postgresql/postgresql.conf.cfengine.patch [hunk 4]
```diff
@@ -50,9 +50,9 @@
 +# actively intend to use prepared transactions.
 +work_mem = 48MB				# min 64kB
  #hash_mem_multiplier = 2.0		# 1-1000.0 multiplier on hash table work_mem
--#maintenance_work_mem = 64MB		# min 1MB
-+maintenance_work_mem = 512MB		# min 1MB
- #autovacuum_work_mem = -1		# min 1MB, or -1 to use maintenance_work_mem
+-#maintenance_work_mem = 64MB		# min 64kB
++maintenance_work_mem = 512MB		# min 64kB
```
Comment (vpodzime) [inline deps-packaging/postgresql/postgresql.conf.cfengine.patch:54]: if the minimum is 1MB, our patched version should say that as well

**Review by vpodzime — COMMENTED** [review]
I think the Perl commits should be squashed. And there are more cherry-pick references to private commits in the commit messages. Looks good to me otherwise.

File: deps-packaging/postgresql/postgresql.conf.cfengine.patch (hunk 4 above)
Comment (craigcomstock) [reply deps-packaging/postgresql/postgresql.conf.cfengine.patch:54]: It is not. The maintenance_work_mem minimum is now 64kB. See this commit: https://github.com/postgres/postgres/commit/bbf668d66fbf61d45f1c187b08a5f51537bfb7c7

File: deps-packaging/openssl/0007-Avoid-using-Interlocked-64-Windows-functions.patch (hunk 3 above)
Comment (craigcomstock) [reply deps-packaging/openssl/0007-Avoid-using-Interlocked-64-Windows-functions.patch:12]: removed.

File: build-scripts/install-dependencies (hunk 2 above)
Comment (craigcomstock) [reply build-scripts/install-dependencies:53]: changed the commit message, now part of openssl commit.

File: deps-packaging/openssl/source (hunk 1 above)
Comment (craigcomstock) [reply deps-packaging/openssl/source:1]: added to openssl bump commit message.

File: deps-packaging/postgresql/postgresql.conf.cfengine.patch (hunk 4 above)
Comment (vpodzime) [reply deps-packaging/postgresql/postgresql.conf.cfengine.patch:54]: This patch of patch is always confusing...

**Review by vpodzime — APPROVED** [review]
ACK

**Pull request merged**